src/communication/rpc.rs

```diff
@@ -91,6 +92,48 @@ pub trait RpcClient: Send + Sync {
     ) -> Result<Option<UPayload>, ServiceInvocationError>;
 }
 
+impl dyn RpcClient {
```

Comment:
Shouldn't this simply be a default implementation in the trait instead?

Reply:
Nope - those have to be statically resolvable at compile-time, which doesn't work with generic types.